### PR TITLE
Four functionality additions to make roots-contentful more flexible

### PR DESCRIPTION
**Pull request by gdeividas**
1. **Added ability to namespace**
    Now you can specify were "content_type's" are attached globally and per content_type, rather than always getting assign to "contentful".

2. **Content_type now has 'set_locals' function**
    Basically ability to set "content_type" locals manually rather than always automatically

3. **Path function now has second parameter passed containing all content types(deep cloned)**
   Very useful when you need 'data' form other content_type for curren type's path generation.., i.e having `meta` content_type, on which other content_type may depend..

4. **Minor execution flow reordering**
    Basically set "urls"/ write files right before compilation, so that `path` fn, has all the data for really custom path generation..

### CODE REVIEW

**Comment by declandewet** [comment]
Hi @gdeividas! First off, thank you very much for these wonderful additions to `roots-contentful`. They'll be very useful and the inclusion of tests is :a:+

Secondly, I would like to apologize for not responding this PR sooner. I've been having a few issues with notifications on Github.

However, `roots-contentful` is undergoing a [major refactor to ES2016](/carrot/roots-contentful/pull/#55) at the moment.

Is there any chance you could open a separate PR that adds these features to the `es2016` branch? I can merge them in there and then they will be included in the next release

**Comment by gdeividas** [comment]
@declandewet sure will do it within next week.
